src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -289,6 +290,13 @@ private boolean matches(Offer offer, Request request) {
     List<Range> ports = null;
 
     for (Resource resource : offer.getResourcesList()) {
+      String resourceRole = resource.getRole();
+      String expectedRole = mesosCloud.getRole();
+      if (! resourceRole.equals(expectedRole)) {
```

Comment:
the plugin should still accept `*` offers, even if set to a specific role.
marathon and chronos are doing the same.

this should be:

    if (! (resourceRole.equals(expectedRole) || resourceRole.equals("*"))) {